Project1/AttackHelper.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "MapHelper.h"
+#include "MoveHelper.h"
+#include <vector>
+#include "GridMap.h"
+#include "MapHelper.h"
+#include "GridCell.h"
+#include "ID.h"
+
+class Unit;
+class GridCell;
+class AttackHelper :
+	private MapHelper
+{
+public:
+	AttackHelper(GridMap& map);
+
+	std::vector<Unit::Ref> getAttackableUnits(const Unit& unit);
+	const std::vector<Unit::Ref> getAttackableUnits(const Unit& unit) const;
```

Comment:
I don't think the constness here makes sense. If you're trying to keep callers of the const method from modifying the units, I don't think it works. They can just create a new vector with all the unit refs in it and modify them from there.

If that's what you want, the const one should return a vector of const Unit::Red

Reply:
I don't think it makes sense to have a non-const version anyway, since this is for combat and attacking is not const. I'll just remove it